provision/kubernetes/minikube/README.markdown

```diff
@@ -0,0 +1,830 @@
+          "tls_disable": true
+        }
+      }],
+      "storage": [{
+        "consul": {
+          "address": "localhost:8500",
+          "schema": "http",
+          "path": "vault"
+        }
+      }]
+    }
+
+---
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vault-consul-local-config
+data:
+  local.config: |
+    {
+      "server": false,
+      "client_addr": "127.0.0.1",
+      "ui": false,
+      "raft_protocol": 3,
+      "retry_join": [
+        "consul-0.consul",
+        "consul-1.consul",
+        "consul-2.consul",
+        "consul-3.consul",
+        "consul-4.consul"
+      ]
+    }
+
+---
+
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: vault
+  labels:
+    app: vault
+    role: server
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: vault
+      role: server
+  template:
+    metadata:
+      labels:
+        app: vault
+        role: server
+    spec:
+      containers:
+      - name: vault
+        image: vault-enterprise
+        imagePullPolicy: Never
+        securityContext:
+          capabilities:
+            add: [ "IPC_LOCK" ]
+        env:
+        - name: VAULT_LOCAL_CONFIG
+          valueFrom:
+            configMapKeyRef:
+              name: vault-local-config
+              key: local.config
+        args: [ "server" ]
+        livenessProbe:
+          exec:
+            command:
+            - pidof
+            - vault
+          initialDelaySeconds: 5
+          timeoutSeconds: 2
+        ports:
+        - containerPort: 8200
+          name: api
+      - name: consul-client
+        image: consul-enterprise
+        imagePullPolicy: Never
+        env:
+          - name: CONSUL_LOCAL_CONFIG
+            valueFrom:
+              configMapKeyRef:
+                name: vault-consul-local-config
+                key: local.config
+        command: [ "docker-entrypoint.sh" ]
+        args: [ "agent", "-config-file", "/consul/config/local.json"]
+        livenessProbe:
+          exec:
+            command:
+            - pidof
+            - consul
+          initialDelaySeconds: 5
+          timeoutSeconds: 2
+        ports:
+        - containerPort: 8500
+          name: api
+```
+
+#### Deploying and Operating
+
+``` example
+$ kubectl apply -f vault.yml
+service "vault-ui" created
+service "vault" created
+configmap "vault-local-config" created
+configmap "vault-consul-local-config" created
+deployment "vault" created
+
+$ kubectl get deployment | grep vault
+vault     3         3         3            3           1m
+
+$ kubectl get pod | grep vault
+vault-5758659497-dsbws   2/2       Running   0          2m
+vault-5758659497-rhc9z   2/2       Running   1          2m
+vault-5758659497-xt2gr   2/2       Running   0          2m
+
+$ minikube service --url vault-ui
+http://192.168.99.100:32148
+
+$ xdg-open $(minikube service --url vault-ui)
+```
+
+##### Init and Unseal
+
+Vault supports the the ability to use a Hardware Security Module(HSM)
```

Comment:
"Vault supports the the ability to use AWS KMS, GCP Cloud KMS, or a Hardware Security Module (HSM) for automated unsealing"